src/lib/logger.ts

```diff
@@ -12,118 +23,65 @@ export const logger: Logger = pino({
     },
   },
   serializers: {
-    req: (req) => {
+    req: (req: Request) => {
       return {
         method: req.method,
-        url: req.url,
+        url: redactQueryParams(req.url),
       };
     },
-    res: (res) => {
+    res: (res: Response) => {
       return {
-        statusCode: res.err?.response?.status || res?.statusCode,
+        statusCode: res.statusCode,
         sessionId: res.locals.sessionId,
+        location: redactQueryParams(res.getHeader("location") as string | undefined),
       };
     },
   },
 });
 
+const addRequestContext = (req: Request, res: Response, val: object): object => ({
+  ...val,
+  requestId: req.id,
+  ipvSessionId: req.session?.ipvSessionId,
+  sessionId: req.session?.id,
+  context: req.session?.context ?? undefined,
+});
```

Comment:
AFAICT pino strips any 'undefined' keys anyway, so I'm not sure we need all the conditional logic.

(If this is not true, we might need to update this)